include/SFML/Window/Cursor.hpp

```diff
@@ -0,0 +1,219 @@
+    ///  sf::Cursor::Help                   |  yes  |    no    |   yes
+    ///  sf::Cursor::NotAllowed             |  yes  |    yes   |   yes
+    ///
+    ////////////////////////////////////////////////////////////
+    enum Type
+    {
+        Arrow,                  ///< Arrow cursor (default)
+        ArrowWait,              ///< Busy arrow cursor
+        Wait,                   ///< Busy cursor
+        Text,                   ///< I-beam, cursor when hovering over a field allowing text entry
+        Hand,                   ///< Pointing hand cursor
+        SizeHorizontal,         ///< Horizontal double arrow cursor
+        SizeVertical,           ///< Vertical double arrow cursor
+        SizeTopLeftBottomRight, ///< Double arrow cursor going from top-left to bottom-right
+        SizeBottomLeftTopRight, ///< Double arrow cursor going from bottom-left to top-right
+        SizeAll,                ///< Combination of SizeHorizontal and SizeVertical
+        Cross,                  ///< Crosshair cursor
+        Help,                   ///< Help cursor
+        NotAllowed              ///< Action not allowed cursor
+    };
+
+public:
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Default constructor
+    ///
+    /// This constructor doesn't actually create the cursor;
+    /// initially the new instance is invalid and must not be
+    /// used until either loadFromPixels() or loadFromSystem()
+    /// is called and successfully created a cursor.
+    ///
+    ////////////////////////////////////////////////////////////
+    Cursor();
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Destructor
+    ///
+    /// This destructor releases the system resources
+    /// associated with this cursor, if any.
+    ///
+    ////////////////////////////////////////////////////////////
+    ~Cursor();
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Create a cursor with the provided image
+    ///
+    /// \a pixels must be an array of \a width by \a height pixels
+    /// in 32-bit RGBA format. If not, this will cause undefined behavior.
+    ///
+    /// If \a pixels is null or either \a width or \a height are 0,
+    /// the current cursor is left unchanged and the function will
+    /// return false.
+    ///
+    /// In addition to specifying the pixel data, you can also
+    /// specify the location of the hotspot of the cursor. The
+    /// hotspot is the pixel coordinate within the cursor image
+    /// which will be located exactly where the mouse pointer
+    /// position is. Any mouse actions that are performed will
+    /// return the window/screen location of the hotspot.
+    ///
+    /// \warning On Unix, the pixels are mapped into a monochrome
+    ///          bitmap: pixels with an alpha channel to 0 are
+    ///          transparent, black if the RGB channel are close
+    ///          to zero, and white otherwise.
```

Comment:
I'd say it's safe to ignore, considering we want to use modern OpenGL anyway? Even my old 1 MB Cirrus Logic from 1996 supported colored/animated hardware cursors, although I remember some later games claiming otherwise. :)